wholesym/src/config.rs

```diff
@@ -1,12 +1,25 @@
-use std::collections::HashMap;
 use std::path::PathBuf;
+use std::{collections::HashMap, sync::Arc};
 
+use debugid::DebugId;
 use symsrv::{parse_nt_symbol_path, NtSymbolPathEntry};
 
+// Helper struct to avoid not being able to derive Debug on SymbolManagerConfig
+pub(crate) struct PrecogDataContainer {
+    pub(crate) precog_data: HashMap<DebugId, Arc<dyn samply_symbols::SymbolMapTrait + Send + Sync>>,
+}
+
+impl std::fmt::Debug for PrecogDataContainer {
+    // Explicit implementation needed due to precog_data
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("...")
+    }
+}
+
 /// The configuration of a [`SymbolManager`](crate::SymbolManager).
 ///
 /// Allows specifying various sources of symbol files.
-#[derive(Debug, Clone, Default)]
```

Comment:
Please keep the config Clone.

Reply:
Ah yes, this can come back. When I was using a trait object in here, I had all sorts of problems but since that's gone this should be fine. It's moving to the helper anyway.